src/client/opamSolution.ml

```diff
@@ -1133,6 +1133,20 @@ let apply ?ask t ~requested ?add_roots ?(assume_built=false) ?force_remove
           action_graph
       in
       let success = match r with | OK _ -> true | _ -> false in
+      let t =
+        if success && (OpamClientConfig.(!r.assume_depexts) ||
+                       OpamStateConfig.(!r.no_depexts)) then
```

Comment:
We don't need to check `no_depexts` as `assume depexts` is defined as `assume_depexts || no_depexts`. But to remove it, we need to be sure that this implication remain always true.